fpfilter.pl

```diff
@@ -564,15 +567,19 @@ sub setup_workdir {
     my $working_bam = File::Spec->catfile($dir, "tumor.bam");
     my $working_bam_index = File::Spec->catfile($dir, "tumor.bam.bai");
     symlink $bam_file, $working_bam;
-    if($bam_index) {
-        symlink $bam_index, $working_bam_index;
-    }
-    elsif(-e $bam_file . ".bai") {
-        symlink $bam_file . ".bai", $working_bam_index;
+
+    unless ($bam_index) {
+        my ($bam_filename,$bam_path,$bam_suffix) = File::Basename::fileparse($bam_file,qw/\.bam/);
```

Comment:
It's effectively the same, but the extension might be better as `qr/\.bam/` instead of `qw/\.bam/` to make clearer it's used as a regex (which explains the `\`).